### PR TITLE
En 5276 added genesis time check in fork detector

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added prints info
* Added request for missing overwritten header when a finality check is done
* Added check for genesis time missmatch in fork detector

### CODE REVIEW

File: process/block/metablock.go [hunk 1]
```diff
@@ -1133,7 +1133,12 @@ func (mp *metaProcessor) checkShardHeadersFinality(highestNonceHdrs map[uint32]d
 		}
 
 		if nextBlocksVerified < mp.shardBlockFinality {
-			go mp.onRequestHeaderHandlerByNonce(lastVerifiedHdr.GetShardID(), lastVerifiedHdr.GetNonce()+1)
+			for nonce := lastVerifiedHdr.GetNonce(); nonce <= lastVerifiedHdr.GetNonce()+1; nonce++ {
```
Comment (sasurobert) [inline process/block/metablock.go]: this is a for for 2 elements, would it be better to have two calls of go mp.onRequestHeaderHandler

File: process/block/shardblock.go [hunk 2]
```diff
@@ -377,7 +377,12 @@ func (sp *shardProcessor) checkMetaHdrFinality(header data.HeaderHandler) error
 	}
 
 	if nextBlocksVerified < sp.metaBlockFinality {
-		go sp.onRequestHeaderHandlerByNonce(lastVerifiedHdr.GetShardID(), lastVerifiedHdr.GetNonce()+1)
+		for nonce := lastVerifiedHdr.GetNonce(); nonce <= lastVerifiedHdr.GetNonce()+1; nonce++ {
+			go func(requestedNonce uint64) {
```
Comment (sasurobert) [inline process/block/shardblock.go]: same as for metachain. 2 calls are easier to understand than this for.

File: process/sync/baseForkDetector.go [hunk 3]
```diff
@@ -61,21 +62,17 @@ func (bfd *baseForkDetector) checkBlockBasicValidity(
...
-			log.Trace("add requested item with error",
-				"error", err.Error(),
-				"key", headerHash)
-
-		}
-
+		process.AddHeaderToBlackList(bfd.blackListHandler, headerHash)
 		return process.ErrHeaderIsBlackListed
 	}
+	if genesisTimeFromHeader != bfd.genesisTime {
```
Comment (sasurobert) [inline process/sync/baseForkDetector.go]: I think this check is better in the interceptor.

File: process/block/metablock.go (hunk 1 above)
Comment (SebastianMarian) [reply process/block/metablock.go]: done

File: process/block/shardblock.go (hunk 2 above)
Comment (SebastianMarian) [reply process/block/shardblock.go]: done

File: process/sync/baseForkDetector.go (hunk 3 above)
Comment (SebastianMarian) [reply process/sync/baseForkDetector.go]: added TODO to remove this check when implementation on interceptors side will be done

**Pull request merged**